website/static/js/pages/profile-settings-addons-page.js

```diff
@@ -89,6 +91,18 @@ for (var i=0; i < addonEnabledSettings.length; i++) {
    }
 }
 
+/* Before closing the page, Check whether the newly checked addon are updated or not */
+$(window).on('beforeunload',function() {
+    //new checked items but not updated
+    var checked = uncheckedOnLoad.filter($('#selectAddonsForm input:checked'));
```

Comment:
That's true! It is not necessary. I have changed it. But I just found this fix had a logic problem, which would invoke the unnecessary warming message when submitting. I will keep working on it!